### PR TITLE
Tramstation: Growing Pains

### PR DESCRIPTION
**Pull request by MMMiracles**
## About The Pull Request

image

## Why It's Good For The Game

Lots of QoL and structural changes in attempt to make the cool map even cooler.

## Changelog
:cl: MMMiracles
add: Tramstation has received a substantial overhaul! Please consult your local tour guide and station maps for changes.  
/:cl:

**Changes (as of so far)**
- The Tramstation tunnels have been extended 6 tiles each way, making that trek across the central rail a little more dangerous.
- No more mid-way safety net on the transit rails. You're either making it or you're jumping to the bottom floor to avoid the tram.
- The central rail apparently had a negative slowdown, meaning you actually WERE faster to just run the gauntlet because it literally made you faster. This has been reverted because I want you to get hit by a train.
- The side routes are now a bit more dangerous since you can get pushed off into the lower floor
- Fauna overhaul! Several locations including the transit tunnels have gotten some greenery to help liven up transit between sectors. Please do not rip up the AstroTurf it is very expensive in space.
- Handicap-accessible departments! Every major wing and departments with multiple floors has been given a 2x3 elevator segment for those among the crew who have been hit by the tram a few too many times. Handicap inaccessible staircases may or may not come after this (i hate the handicapped)
- Expanded Security wing! You know that lame hallway between interrogation and the courtroom access? Now its a whole holding wing fit with essentials to keep your inmates content while waiting for their due process (never ever).
- Reworked Bridge! Front row seats to the western tram dock as well as a furnished meeting room with various corporate memorabilia!
- The HoP's office has been moved to function more as an arrival gate for late joining crew! Scenic queue lines and an option to block off the lower dorm access if you really want to enforce the 'Papers, Please' roleplay you've always wanted out of your HoP experience.
- Combined the teleporter/gateway/eva access into one special external operations room. All you off-station needs in one convenient place! 
- More multi-z integration! Several segments (mainly maintenance level transfers) have been given overhangs and better methods to move between levels. This is still being expanded upon as of current PR posting.
- The power/pipe access shafts have been changed to be more public-facing and now also act as another inbetween for maintenance-dwelling crew to shift between floors.
- Multi-z solars! Both solar wings have been extensively overhauled to include multi-z structuring and easily doubled the amount of roundstart panels on the map. 
- Escape pod bay! [Casually borrowing from a certain ship map](https://tgstation13.org/phpBB/viewtopic.php?t=32834), there is now a centralized location for all station escape pods on the floor below Arrivals. Honestly makes more sense thematically instead of having a bunch of scattered bays.
- MULEBOT integration! Each major department now has delivery drop-off points along with Cargo getting a minor restructuring to fit 4 MULEBOTs. Seedy side-tunnels and drop points have been added for departments that aren't normally accessible from upper maintenance so they can still properly deliver cargo if sent this way. I can't guarantee this won't end in MULEBOTs getting ran over by a tram because they're fickle beasts.
- Complete rework of the disposals/piping/wirenet! I literally ripped everything up and rebuilt it with (hopefully) better stability in mind. Disposals is now also set up in that it'll try to avoid going through departments unnecessarily if your package isn't marked for it.
- Cargo's mail room and trash room has been overhauled to be more easily accessed for cargo techs and for routing mail.
- The chef has access to the morgue's back door via the front maintenance hatch while Robotics can now access the same back door via their maintenance shaft.
- The chem lab's starting area has been expanded so chemists don't have to worry as much about digging if they don't have large projects.

### CODE REVIEW

**Comment by Time-Green** [comment]
This is so fucking cool! MORE PICTURES!!!

**Comment by MMMiracles** [comment]
Updated with xbox-huge images of both floors

**Comment by Serxule** [comment]
2 things,
1. why are there so cycling airlocks when they dont lead anywhere with different atmos? (before they were added as the side tunnels had no atmos)
2. is there no firelocks in lower tram?

**Comment by san7890** [comment]
https://github.com/tgstation/tgstation/milestone/62?closed=1 should have a bunch of things that you may have missed

**Comment by MMMiracles** [comment]
> 2 things,
> 
>     1. why are there cycling airlocks when they dont lead anywhere with different atmos? (before they were added as the side tunnels had no atmos)
> 
>     2. is there no firelocks in lower tram?

1. makes it take longer
2. what do you mean?

**Comment by Serxule** [comment]
> 1. makes it take longer
> 2. what do you mean?

1. that just makes people like the map less
2. i dont see any firelocks in lower tram

**Comment by MrDoomBringer** [comment]
the tram tunnel being longer is huge, excellent change

**Comment by CMDR-Gungnir** [comment]
based as fuck, i like it

**Comment by RuneQuester** [comment]
No more safe spots on the rail, and the track is longer. 

Why. 

Stop forcing the clunky ass tram into our faces as the only way of upper z level travel

**Comment by MMMiracles** [comment]
> No more safe spots on the rail, and the track is longer.
> 
> Why.
> 
> Stop forcing the clunky ass tram into our faces as the only way of upper z level travel

if you needed the safety spot you're a coward and deserve getting hit for half-assing the tram gauntlet

**Comment by lessthnthree** [comment]
Can't wait!

Here's a bunch of PRs, not sure which you've implemented.
https://github.com/tgstation/tgstation/pulls?q=is%3Apr+author%3Alessthnthree+

**Comment by BlueMemesauce** [comment]
I disliked tramstation before because those side hallways were just more convenient than using the tram, which made the tram feel like an gimmick to me. I was indecisive about it before, but I think making the tunnel longer and nerfing the side hallways is a great change that will make using the tram feel much more practical.

**Comment by Jacquerel** [comment]
Unless something happened in the backend that I missed the tram still can’t carry air, that implies that those tunnels are pressurised if that is what gave you the idea that it can.

File: code/game/area/areas/station.dm [hunk 1]
```diff
@@ -1408,6 +1408,20 @@
 	icon_state = "abandoned_sci"
 	sound_environment = SOUND_AREA_SMALL_ENCLOSED
 
+/area/station/escapepodbay
+	name = "\improper Pod Bay"
+	icon_state = "podbay"
+
+/area/station/asteroid
+	name = "\improper Station Asteroid"
+	icon_state = "station_asteroid"
```
Comment (lessthnthree) [inline code/game/area/areas/station.dm:1417]: The icons don't start with station_
It would just be icon_state = "asteroid"

**Comment by Pinkufeck** [comment]
Yet another update that will make upper and lower Z level walkways shit. Upper Z level still lacks maints. Lower Z level is still mostly composed of very unimportant things expect for engineering and telecomms which will still explode together when one does.  Also multi Z level solars seem particularly annoying to setup and way less dense than, not that they get set anyway.

Overall it goes with the rest of updates you made which aim at forcing people to use the tram, levels despite it being easily replacable by just, walking.

If you want to push people to use the tram, make it be faster than walking, like, much faster, and space stuff more, what you are doing is just creating an issue and offering it's solution.

**Comment by Pinkufeck** [comment]
Also, don't know if you are going to change that or not, but the maintenance tunnels are way too linear and lack rooms.

**Comment by Pinkufeck** [comment]
Also, while thematically yes it makes sense to centralize pods it poses some issues. What about antagonists that want to evac ? What about prisonner extraction from perma (even if it's less of an issue) ?

**Comment by MMMiracles** [comment]
> Yet another update that will make upper and lower Z level walkways shit. Upper Z level still lacks maints. Lower Z level is still mostly composed of very unimportant things expect for engineering and telecomms which will still explode together when one does. Also multi Z level solars seem particularly annoying to setup and way less dense than, not that they get set anyway.

- Walkways are still just as before, the only main difference being taking the stupid route is more dangerous.
- I'm not sure exactly what you'd want out of maintenance that is designed to connect between segments, unless you're expecting some sprawling labyrinth of random storage rooms and garbage loot to sift through.
- Toxins, Xenobiology, Mining, the whole actual brig, and Hydroponics is unimportant? I don't get what you're getting at here, the lower floor acts as an extension of the upper floor and apart from Service, isn't meant to be a public hallway.
- If you don't want to lose comms, maybe don't let the giant crystal explode? You're acting like comms exploding is a result of a minor nuisance and not something that usually results in a shuttle call anyway. Be lucky nobody has managed multi-z explosions yet because Medical going out with the engine will be 100% intentional if/when it gets added.
- Local testing (from my own experience) makes solar setup about as much effort as before, if not slightly more interesting than just running in a line and placing cable occasionally. Looks cooler in-game anyway.

> Overall it goes with the rest of updates you made which aim at forcing people to use the tram, levels despite it being easily replacable by just, walking, and it doesn't offer any solution for big issues like SM and telecomms being too close. If you want to push people to use the tram, make it be faster than walking, like, much faster, and space stuff more, what you are doing is just creating an issue and offering it's solution.

- Between my own financial and personal time invested on top of the work @lessthnthree has put in during my time spent on this rework has been mainly towards tram QoL (!!literally made it faster!!, quicker departures/arrivals, less down time between uses, faster doors, ect), this project has very been both a carrot and stick approach to encouraging use of the tram.
- crystal go boom lol
- Already covered the walking segment (it is reasonably faster than walking and is literally the most direct method between sides quickly). This PR also literally spaces things out more so I'm not sure why this is brought up.

> Also, don't know if you are going to change that or not, but the maintenance tunnels are way too linear and lack rooms.

If you had read the PR you wouldn't of brought this up.

**Comment by Pinkufeck** [comment]
> * Walkways are still just as before, the only main difference 
Still just as before, but they have multiple differences, so they aren't the same, are they.

> * I'm not sure exactly what you'd want out of maintenance that is designed to connect between segments, unless you're expecting some sprawling labyrinth of random storage rooms and garbage loot to sift through.
Somewhere for antags to get into departement from the upper z level without being forced to get eva gear. Cus that's been an issue

> * Toxins, Xenobiology, Mining, the whole actual brig, and Hydroponics is unimportant? I don't get what you're getting at here, the lower floor acts as an extension of the upper floor and apart from Service, isn't meant to be a public hallway.

Let's say, you singulo. In most cases you do that, toxins, already done or won't be. "the whole actual brig" aka just perma and only perma, isn't important. xenobio isn't done most rounds, and hydroponics really doesn't contribute to nothing in 90% of cases either. 
My point is, we have a lower Z level, where the only full departement that's important, will already be crippled by a Sm explosion. So singulo is pointless, you don't threaten anyone more than if you just blew it up. 

> * If you don't want to lose comms, maybe don't let the giant crystal explode? You're acting like comms exploding is a result of a minor nuisance and not something that usually results in a shuttle call anyway. Be lucky nobody has managed multi-z explosions yet because Medical going out with the engine will be 100% intentional if/when it gets added.

comms can be fixed, quite easily. one of the traitor objs is "destroy comms hub". That's a case in which, you'll fix comms and maybe not call. My point is, in other maps, telecomms is somewhat secure. Here a simple SM sabotage bypasses the need to go out of your way to cut comms, makes the SM and tcomm two bird with one stone. If you blow the SM on any other map, for a little while, you'll still be able to organize with others

> * Local testing (from my own experience) makes solar setup about as much effort as before, if not slightly more interesting than just running in a line and placing cable occasionally. Looks cooler in-game anyway.
>
"It's good because i said it was and placing cables in straight AND diagonal lines is more stimulating than cables in a line, and it doesn't matter i think it looks bettee"
> 
> * Between my own financial and personal time invested on top of the work @lessthnthree has put in during my time spent on this rework has been mainly towards tram QoL (!!literally made it faster!!, quicker departures/arrivals, less down time between uses, faster doors, ect), this project has very been both a carrot and stick approach to encouraging use of the tram.
>
I'm gonna be really harsh here, but just because you invested time into an idea  doesn't mean  it's good or worth implementing
> 
> If you had read the PR you wouldn't of brought this up.

was below the screenshots, missed it, shit happens.

**Comment by Pinkufeck** [comment]
Pushing people towards using a centralized transportation  that you makes you wait by making any other altnertives trash doesn't make the map fun to play in.
Consider a game in which to go from point a to point b you can either : 
A) use a path that goes through several steps and wait at each of those steps to continue, there's no reason to wait at those steps except a timer put in by devs to push to solution B
B) Wait longer for someone to move you 

Most people are gonna go A, and they are gonna feel frustrated, because there's no reason for those timers but to justify way B. Sure some people will enjoy B, because maybe they talk to the person, but that won't be most people, because people will go towards the efficient route. 

I think you should outright remove the walkways. Commit fully to the tram, and stop keeping ways that you are slowly deprecating into being the most annoying route possible.

**Comment by CMDR-Gungnir** [comment]
> Pushing people towards using a centralized transportation that you makes you wait by making any other altnertives trash doesn't make the map fun to play in. Consider a game in which to go from point a to point b you can either : A) use a path that goes through several steps and wait at each of those steps to continue, there's no reason to wait at those steps except a timer put in by devs to push to solution B B) Wait longer for someone to move you
> 
> Most people are gonna go A, and they are gonna feel frustrated, because there's no reason for those timers but to justify way B. Sure some people will enjoy B, because maybe they talk to the person, but that won't be most people, because people will go towards the efficient route.
> 
> I think you should outright remove the walkways. Commit fully to the tram, and stop keeping ways that you are slowly deprecating into being the most annoying route possible.

Tram Good.

**Comment by MMMiracles** [comment]
> > * Walkways are still just as before, the only main difference
> 
> Still just as before, but they have multiple differences, so they aren't the same, are they.

The gauntlet was made longer and no safety net, the side routes take slightly longer and I guess there's now the threat of being pushed off by someone else? Maintenance is very much still just maintenance.

> > * I'm not sure exactly what you'd want out of maintenance that is designed to connect between segments, unless you're expecting some sprawling labyrinth of random storage rooms and garbage loot to sift through.
> 
> Somewhere for antags to get into departement from the upper z level without being forced to get eva gear. Cus that's been an issue

Every department has a maintenance hatch on top of being accessible through just literally mining the rock with one of the plethora of pickaxes strewn about maintenance. Literally what department are you needing EVA gear to access? Tram has used porus rocks for months now and hasn't required any external gear for digging.

> > * Toxins, Xenobiology, Mining, the whole actual brig, and Hydroponics is unimportant? I don't get what you're getting at here, the lower floor acts as an extension of the upper floor and apart from Service, isn't meant to be a public hallway.
> 
> Let's say, you singulo. In most cases you do that, toxins, already done or won't be. "the whole actual brig" aka just perma and only perma, isn't important. xenobio isn't done most rounds, and hydroponics really doesn't contribute to nothing in 90% of cases either. My point is, we have a lower Z level, where the only full departement that's important, will already be crippled by a Sm explosion. So singulo is pointless, you don't threaten anyone more than if you just blew it up.

This is a multi-z issue with station-destroying elements still not taking into account other floors. You'd see this same 'issue' on any map that was actually built with multi-z in mind. 

> > * If you don't want to lose comms, maybe don't let the giant crystal explode? You're acting like comms exploding is a result of a minor nuisance and not something that usually results in a shuttle call anyway. Be lucky nobody has managed multi-z explosions yet because Medical going out with the engine will be 100% intentional if/when it gets added.
> 
> comms can be fixed, quite easily. one of the traitor objs is "destroy comms hub". That's a case in which, you'll fix comms and maybe not call. My point is, in other maps, telecomms is somewhat secure. Here a simple SM sabotage bypasses the need to go out of your way to cut comms, makes the SM and tcomm two bird with one stone. If you blow the SM on any other map, for a little while, you'll still be able to organize with others

Again, correlating the SM being sabotaged and allowed to go critical isn't exactly a 'small' issue. Genuinely baffled by this complaint and it sounds more like some weird theorycraft than an actual issue.

> > * Local testing (from my own experience) makes solar setup about as much effort as before, if not slightly more interesting than just running in a line and placing cable occasionally. Looks cooler in-game anyway.
> 
> "It's good because i said it was and placing cables in straight AND diagonal lines is more stimulating than cables in a line, and it doesn't matter i think it looks bettee"

thank you for conceding your point

> > * Between my own financial and personal time invested on top of the work @lessthnthree has put in during my time spent on this rework has been mainly towards tram QoL (!!literally made it faster!!, quicker departures/arrivals, less down time between uses, faster doors, ect), this project has very been both a carrot and stick approach to encouraging use of the tram.
> 
> I'm gonna be really harsh here, but just because you invested time into an idea doesn't mean it's good or worth implementing

My issue lies in that all the apparent issues you bring up have either been recently addressed or are literally being addressed in this PR. It makes me question the validity of these issues if I can't be certain you've even actually played the map recently.

> > If you had read the PR you wouldn't of brought this up.
> 
> was below the screenshots, missed it, shit happens.

bruh if you gonna walk in starting shit you better at least be reading the whole PR before starting something

**Comment by Zergspower** [comment]
Hey as long as we get actual maintenance (with welding fuel omg) i'm sold, and perhaps getting rid of the 2nd circuit room because i think someone thought there wasnt one...

**Comment by Zergspower** [comment]
image
 oh it's still there, when the 2nd one was placed we lost the abandoned robo bay

**Comment by MMMiracles** [comment]
> Does the tram still break if you move the target slightly then put everyone on it in parallel substep on it?

im gonna be real with you man i have no idea what you just said

**Comment by MMMiracles** [comment]
Maintenance is finally done and apart from me still going through testing disposals, should be ready to start peering through the absolute mess of map files.

**Comment by DAKKA-WAAAGH** [comment]
Tramstation part 3: we fucking deleted it lmao

Also great job king, you're doing amazing! 😄

**Review by Maurukas — COMMENTED** [review]
First pass here:
Something is glitched about the east station. The tram can't be summoned there and it does not show up in the tram control console:
image
It looks like the right_part marker isn't placed.

<details>
<summary>Missing APCs in the following areas: </summary>
These generally have modular areas, but there are often doors in them that are infinitely powered as a result of the missing APC. Some others are station areas proper

* /area/station/hallway/secondary/construction/engineering
* /area/station/maintenance/starboard/lesser
* /area/station/maintenance/department/eva
* /area/station/maintenance/port/aft
</details>
<details>

<summary>Missing air alarms in the following areas (that need them) </summary>
/area/station/hallway/primary/tram/left
/area/station/hallway/primary/tram/center
/area/station/command/gateway

</details>

**Review by Maurukas — COMMENTED** [review]
This modular segment near dorms extends out further than the ceiling above it, resulting in roundstart ATs:
image

**Comment by IndieanaJones** [comment]
I'll admit that I think Tram on live is a shitty map. Not very exciting, exists only to have people get hit by the Tram, and it generally depopulates the server. However, seeing this PR testmerged actually changed my opinion on it, as this PR adds a lot of cool shit to find and explore and just a lot of soul in general. I still think the tram itself is lame, but this map now boasts a large amount of redeeming qualities that actually makes it quite fun to explore and navigate. This is an excellent PR and I'll be extremely happy to see the day when I don't loathe going to Tramstation.

On another note, RD's office and robotics don't start connected to the electrical grid. Please fix, thanks.

**Pull request merged**